src/utils/liquidityPools.js

```diff
@@ -269,10 +302,23 @@ export const getRemoveLiquidityTransactions = async (
     symbol: ETH,
   }];
 
-  const erc20Contract = getContract(poolToken.address, ERC20_CONTRACT_ABI);
-  const approvedAmountBN = erc20Contract
-    ? await erc20Contract.allowance(sender, UNISWAP_ROUTER_ADDRESS)
-    : null;
+  const erc20Contract = etherspotService.getContract<?EtherspotErc20Interface>(
+    CHAIN.ETHEREUM,
+    ERC20_CONTRACT_ABI,
+    poolToken.address,
+  );
+
+  if (!erc20Contract) {
+    reportErrorLog('getRemoveLiquidityTransactions: no erc20Contract', { poolToken });
+    return removeLiquidityTransactions;
+  }
+
+  let approvedAmountBN;
+  try {
+    approvedAmountBN = await erc20Contract.callAllowance(sender, UNISWAP_ROUTER_ADDRESS);
+  } catch (error) {
+    reportErrorLog('getRemoveLiquidityTransactions failed: callAllowance failed', { poolToken, error });
+  }
 
   if (!approvedAmountBN || poolTokenAmountBN.gt(approvedAmountBN)) {
```

Comment:
Seems like `callAllowance` return type is now ethers.js BigNumber. Does it still work correctly with regular BigNumber?

Reply:
Please also verify other places

Reply:
It was ethers BigNumber before as well since it was using ethers Contract interface directly and now the return is wrapper through Etherspot SDK.

Reply:
My suggestion would be to transform it to `bignumber.js` e.g. using `fromEthersBigNumber`. Otherwise were working on `any` type and mixing two not really compatible types.

Reply:
Not sure if we're on same track mate. Two things:
1. `fromEthersBigNumber` is used to convert wei values in that are in ethers BigNumber to other BigNumber lib.
2. ethers BigNumber was used in these calls before and replacing each usage to other BigNumber lib is out of scope for this PR as each use has different approaches.